### PR TITLE
clean up a bit

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Remove redundant API methods in VelocyPackHelper, use StaticStrings more consistently
Enterprise companion PR: https://github.com/arangodb/enterprise/pull/356

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] Strictly *new functionality* (i.e. a new feature / new option, no need for porting)

### Testing & Verification

This change is a trivial rework / code cleanup without any test coverage.

https://172.16.10.101/view/PR/job/arangodb-matrix-pr/7099/

### CODE REVIEW

File: arangod/Sharding/ShardingInfo.cpp [hunk 1]
```diff
@@ -45,15 +45,15 @@ using namespace arangodb;
 
 ShardingInfo::ShardingInfo(arangodb::velocypack::Slice info, LogicalCollection* collection)
     : _collection(collection),
-      _numberOfShards(basics::VelocyPackHelper::readNumericValue<size_t>(info, StaticStrings::NumberOfShards,
+      _numberOfShards(basics::VelocyPackHelper::getNumericValue<size_t>(info, StaticStrings::NumberOfShards,
                                                                          1)),
       _replicationFactor(1),
       _writeConcern(1),
       _distributeShardsLike(basics::VelocyPackHelper::getStringValue(info, StaticStrings::DistributeShardsLike,
                                                                      "")),
```
Comment (dothebart) [inline arangod/Sharding/ShardingInfo.cpp]: VPackStringRef() here too?

File: arangod/VocBase/LogicalCollection.cpp [hunk 2]
```diff
@@ -111,7 +111,7 @@ arangodb::LogicalDataSource::Type const& readType(arangodb::velocypack::Slice in
   static const auto& unknown =
       arangodb::LogicalDataSource::Type::emplace(arangodb::velocypack::StringRef(""));
```
Comment (dothebart) [inline arangod/VocBase/LogicalCollection.cpp]: "" not needed?

File: arangod/VocBase/LogicalCollection.cpp [hunk 3]
```diff
@@ -137,21 +137,21 @@ LogicalCollection::LogicalCollection(TRI_vocbase_t& vocbase, VPackSlice const& i
           ::readStringValue(info, StaticStrings::DataSourceName, ""), planVersion,
```
Comment (dothebart) [inline arangod/VocBase/LogicalCollection.cpp]: VPackStringRef() here too?

File: arangod/VocBase/LogicalCollection.cpp [hunk 4]
```diff
@@ -137,21 +137,21 @@ LogicalCollection::LogicalCollection(TRI_vocbase_t& vocbase, VPackSlice const& i
...
-      _isSmart(Helper::readBooleanValue(info, StaticStrings::IsSmart, false)),
+      _isSmart(Helper::getBooleanValue(info, StaticStrings::IsSmart, false)),
 #endif
-      _waitForSync(Helper::readBooleanValue(info, StaticStrings::WaitForSyncString, false)),
-      _allowUserKeys(Helper::readBooleanValue(info, "allowUserKeys", true)),
+      _waitForSync(Helper::getBooleanValue(info, StaticStrings::WaitForSyncString, false)),
+      _allowUserKeys(Helper::getBooleanValue(info, "allowUserKeys", true)),
 #ifdef USE_ENTERPRISE
       _smartJoinAttribute(
           ::readStringValue(info, StaticStrings::SmartJoinAttribute, "")),
```
Comment (dothebart) [inline arangod/VocBase/LogicalCollection.cpp]: VPackStringRef() here too?

File: arangod/VocBase/LogicalDataSource.cpp [hunk 5]
```diff
@@ -170,7 +170,7 @@ LogicalDataSource::LogicalDataSource(Category const& category, Type const& type,
           basics::VelocyPackHelper::getStringValue(definition, StaticStrings::DataSourceName,
                                                    ""),
```
Comment (dothebart) [inline arangod/VocBase/LogicalDataSource.cpp]: VPackStringRef() here too?

File: arangod/VocBase/LogicalCollection.cpp (hunk 2 above)
Comment (jsteemann) [inline arangod/VocBase/LogicalCollection.cpp]:
```suggestion
      arangodb::LogicalDataSource::Type::emplace(arangodb::velocypack::StringRef());
```

**Pull request merged**